storage/backup-restore/src/adapter/local_storage/mod.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#[cfg(test)]
+mod local_storage_test;
+
+use super::Adapter;
+use crate::FileHandle;
+use anyhow::Result;
+use async_trait::async_trait;
+use futures::{stream::BoxStream, StreamExt};
+use rand::RngCore;
+use std::{
+    io::{BufRead, BufReader, Write},
+    path::{Path, PathBuf},
+};
+
+const FILENAME_LEN: usize = 16;
+
+/// A storage backend that stores everything in a local directory.
+pub struct LocalStorage {
+    /// The path where everything is stored.
+    dir: PathBuf,
+}
+
+impl LocalStorage {
+    pub fn new(dir: PathBuf) -> Self {
+        Self { dir }
+    }
+}
+
+#[async_trait]
+impl Adapter for LocalStorage {
+    async fn write_new_file(
+        &self,
+        mut content: impl StreamExt<Item = Vec<u8>> + Send + Unpin + 'async_trait,
```

Comment:
Why is `Unpin` needed?

Reply:
[next](https://docs.rs/futures/0.3.1/futures/stream/trait.StreamExt.html#method.next) requires `Unpin`.